### PR TITLE
Pullquote block: use hooks + API v2

### PR DESCRIPTION
**Pull request by ellatrix**
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/pullquote/edit.js [hunk 1]
```diff
@@ -77,21 +69,39 @@ class PullQuoteEdit extends Component {
 		}
 	}
 
-	pullQuoteTextColorSetter( colorValue ) {
-		const { setTextColor } = this.props;
+	function pullQuoteTextColorSetter( colorValue ) {
 		setTextColor( colorValue );
 		this.wasTextColorAutomaticallyComputed = false;
```
Comment (youknowriad) [inline packages/block-library/src/pullquote/edit.js:74]: Shouuld we use a ref for this?

Comment (ellatrix) [reply packages/block-library/src/pullquote/edit.js:74]: Yes, looks like I forgot some of `this.*` :)

**Review by youknowriad — APPROVED** [review]

**Review by ntsekouras — APPROVED** [review]
LGTM

**Comment by ellatrix** [comment]
Thanks for checking!

This is the only block we have that derives some data from the block's className, which must now be retrieved from `useBlockProps()` instead of a `className` prop on the edit component.

**Comment by ZebulanStanphill** [comment]
> This is the only block we have that derives some data from the block's className, which must now be retrieved from `useBlockProps()` instead of a `className` prop on the edit component.

This feels like a code smell to me. I think stuff like `isSolidColorStyle` should be calculated based on dedicated attributes like `mainColor` and `textColor`, not CSS class name string parsing.

**Pull request merged**